In [11]:
import numpy as np
import pandas as pd

Задание 6.1

In [2]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [7]:
# Ковариация по формуле covXY=M[(X−M(X))(Y−M(Y))]=M(XY)−M(X)M(Y)
covXY = np.mean(zp*ks)-np.mean(zp)*np.mean(ks)
covXY

9157.839999999997

In [5]:
np.cov(zp,ks,ddof = 0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [8]:
# Коэффициент корреляции Пирсона
σX = np.std(zp)
σY = np.std(ks)
r = covXY/(σX*σY)
r

0.8874900920739158

In [14]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [27]:
X = pd.Series(zp)
Y = pd.Series(ks)
X.corr(Y)

0.8874900920739162

Задание 6.2

In [28]:
IQ = [131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
n = 10
X_mean = np.mean(IQ)
X_mean

118.1

In [29]:
σ = np.std(IQ,ddof=1)
σ

10.54566788359614

In [31]:
# по таблице Стьюдента 
t= 2.262

X1 = X_mean + t * σ/ (n ** 0.5)
X1

125.64339223691834

In [32]:
X2 = X_mean - t * σ/ (n ** 0.5)
X2

110.55660776308164

Ответ: С вероятностью 0.95 можно считать, что IQ студентов находится в интервале от 110.55 до 125.64

Задание 6.3

In [38]:
X_mean = 174.2
α = 0.05
n = 27
D = 25
σ = 5
t = 2.055

In [39]:
X1 = X_mean + t * σ/ (n ** 0.5)
X1

176.17742467197445

In [40]:
X2 = X_mean - t * σ/ (n ** 0.5)
X2

172.22257532802553

Ответ: С вероятностью 0.95 можно считать, что рост футболистов находится в интервале от 172,2 до 176,2см